src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java

```diff
@@ -1,47 +1,105 @@
-    public static final Block blockRack                   = new BlockMinecoloniesRack();
-    public static final Block blockWayPoint               = new BlockWaypoint();
+    public static BlockHutTownHall            blockHutTownHall;
+    public static BlockHutCitizen             blockHutCitizen;
+    public static BlockHutMiner               blockHutMiner;
+    public static BlockHutLumberjack          blockHutLumberjack;
+    public static BlockHutBaker               blockHutBaker;
+    public static BlockHutBuilder             blockHutBuilder;
+    public static BlockHutDeliveryman         blockHutDeliveryman;
+    public static BlockHutBlacksmith          blockHutBlacksmith;
+    public static BlockHutStonemason          blockHutStonemason;
+    public static BlockHutFarmer              blockHutFarmer;
+    public static BlockHutFisherman           blockHutFisherman;
+    public static BlockSubstitution           blockSubstitution;
+    public static BlockSolidSubstitution      blockSolidSubstitution;
+    public static BlockHutField               blockHutField;
+    public static BlockHutGuardTower          blockHutGuardTower;
+    public static BlockHutWareHouse           blockHutWareHouse;
+    public static BlockConstructionTape       blockConstructionTape;
+    public static BlockConstructionTapeCorner blockConstructionTapeCorner;
+    public static BlockMinecoloniesRack       blockRack;
+    public static BlockTimberFrame            blockTimberFrame;
+    public static BlockWaypoint               blockWayPoint;
 
 
     // Deactivated for now
-    // public static final Block blockBarrel        = new BlockBarrel();
+    // public static  Block blockBarrel        = new BlockBarrel();
 
     /**
-     * private constructor to hide the implicit public one.
+     * Private constructor to hide the implicit public one.
      */
     private ModBlocks()
     {
+        /*
+         * Intentionally left empty.
+         */
+    }
+
+    /**
+     * Initates all the blocks. At the correct time.
+     */
+    public static void init(final IForgeRegistry<Block> registry)
+    {
+        blockConstructionTape = new BlockConstructionTape().registerBlock(registry);
+        blockConstructionTapeCorner = new BlockConstructionTapeCorner().registerBlock(registry);
+        blockHutBaker = (BlockHutBaker) new BlockHutBaker().registerBlock(registry);
+        blockHutBlacksmith = (BlockHutBlacksmith) new BlockHutBlacksmith().registerBlock(registry);
+        blockHutBuilder = (BlockHutBuilder) new BlockHutBuilder().registerBlock(registry);
+        blockHutCitizen = (BlockHutCitizen) new BlockHutCitizen().registerBlock(registry);
+        blockHutDeliveryman = (BlockHutDeliveryman) new BlockHutDeliveryman().registerBlock(registry);
+        blockHutFarmer = (BlockHutFarmer) new BlockHutFarmer().registerBlock(registry);
+        blockHutField = new BlockHutField().registerBlock(registry);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Synchronize this lazy initialization of 'blockHutField' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2444)